### PR TITLE
add winmd 1.0.210629.2

### PR DESCRIPTION
**Pull request by dvirtz**
Specify library name and version:  **winmd/1.0.210629.2**

Dependency of [cppwinrt](https://github.com/microsoft/cppwinrt)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/winmd/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+sources:
+  # Newer versions at the top
```
Comment (prince-chrismc) [inline recipes/winmd/all/conandata.yml:2]:
```suggestion
```

File: recipes/winmd/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+from conan.tools.microsoft import is_msvc
+
+required_conan_version = ">=1.52.0"
+
+
+class WinMDConan(ConanFile):
+    name = "winmd"
+    description = "C++ winmd parser"
+    # Use short name only, conform to SPDX License List: https://spdx.org/licenses/
+    # In case not listed there, use "LicenseRef-<license-file-name>"
```
Comment (prince-chrismc) [inline recipes/winmd/all/conanfile.py:18]: Please remove the template comments 🙏 

```suggestion
```

File: recipes/winmd/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
```
Comment (prince-chrismc) [inline recipes/winmd/all/conanfile.py:36]: There's a new helper for the migrations with the new msvc compiler

https://github.com/conan-io/conan-center-index/blob/75655c82d50b3c6e57c4cf79a42b277b181083d4/docs/package_templates/cmake_package/all/conanfile.py#L80

File: recipes/winmd/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES C) # if the project is pure C
+project(test_package LANGUAGES CXX) # if the project uses c++
```
Comment (prince-chrismc) [inline recipes/winmd/all/test_package/CMakeLists.txt:3]: More cleanup form tempalte copy pasting

```suggestion
project(test_package LANGUAGES CXX)
```

File: recipes/winmd/all/test_v1_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES C) # if the project is pure C
+project(test_package LANGUAGES CXX) # if the project uses c++
```
Comment (prince-chrismc) [inline recipes/winmd/all/test_v1_package/CMakeLists.txt:3]:
```suggestion
project(test_package LANGUAGES CXX)
```

**Review by prince-chrismc — COMMENTED** [review]
Is this windows only? seems like it might be on the surface?

Just a few questions -- also I noticed I need to update the header only template so there's a few extras in there

**Comment by dvirtz** [comment]
> Is this windows only? seems like it might be on the surface?
Although it's used for building Windows applications, I don't see why to limit it if it builds on every platform.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/winmd/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+required_conan_version = ">=1.52.0"
+
+
+class WinMDConan(ConanFile):
+    name = "winmd"
+    description = "C++ winmd parser"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/winmd"
+    topics = ("native", "C++", "WinRT", "WinMD")
```
Comment (uilianries) [inline recipes/winmd/all/conanfile.py:20]:
```suggestion
    topics = ("native", "cpp", "winrt", "winmd")
```
Prefer lower-case always, and avoid symbols. It should consider search engine when using topics.

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**